Index: pinball-0.3.1/pinball.desktop
===================================================================
--- pinball-0.3.1.orig/pinball.desktop	2009-10-13 18:36:40.000000000 +0200
+++ pinball-0.3.1/pinball.desktop	2009-10-13 18:36:44.000000000 +0200
@@ -1,12 +1,9 @@
 [Desktop Entry]
-Encoding=UTF-8
-Categories=Application;Game;ArcadeGame;X-Red-Hat-Extra
+Categories=Game;ArcadeGame;X-Red-Hat-Extra;
 X-Desktop-File-Install-Version=0.2
 Name=Emilia Pinball
 Comment=OpenGL pinball game
 Exec=pinball
 Icon=pinball.xpm
-Terminal=0
+Terminal=false
 Type=Application
-                                                                                                                                                             
-                                                                                                                                                             
