From: Innocent De Marchi <tangram.peces@gmail.com>
Date: Fri, 31 Jul 2020 12:18:31 +0200
Subject: Fix constant.

Last-Update: 2018-01-21
---
 base/Config.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base/Config.cpp b/base/Config.cpp
index 11ed53c..aafaf55 100755
--- a/base/Config.cpp
+++ b/base/Config.cpp
@@ -416,7 +416,7 @@ void Config::setPaths(char const * const argv0) {
   m_sDataDir = string(EM_DATADIR) + "/";
   m_sExeDir = "./";
   if ( *( m_sDataDir.c_str() ) != '/' ) {
-    char* ptr=0; 
+    const char* ptr=0;
     char* ptrw = 0;
     //cout<<"relative to exe file"<<endl;
     ptr = (strrchr(argv0,'/')); // unix /cygwin / check win32 
