From: Innocent De Marchi <tangram.peces@gmail.com>
Date: Fri, 31 Jul 2020 12:18:31 +0200
Subject: Eliminated the exit with the insert key.

Bug-Debian: http://bugs.debian.org/450763
Last-Update: 2018-03-16
---
 src/Pinball.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/Pinball.cpp b/src/Pinball.cpp
index c98a4ed..f128373 100755
--- a/src/Pinball.cpp
+++ b/src/Pinball.cpp
@@ -742,7 +742,8 @@ int main(int argc, char *argv[]) {
     engine->resetTick();
 
     while (! (  SDL_QuitRequested()  //cout<<"catch close win"<<endl; //!rzr
-                || Keyboard::isKeyDown(SDLK_INSERT)))  {
+//               || Keyboard::isKeyDown(SDLK_INSERT)
+		))  {
 #if EM_DEBUG
       if (Keyboard::isKeyDown(SDLK_p)) {
         Keyboard::waitForKey();
