Description: Eliminated the exit with the insert key.
Author: Innocent De Marchi <tangram.peces@gmail.com>
Bug-Debian: http://bugs.debian.org/450763
Last-Update: 2018-03-16

--- pinball-0.3.1.orig/src/Pinball.cpp	2003-11-20 17:46:17.000000000 +0100
+++ pinball-0.3.1/scr/Pinball.cpp	2018-03-16 19:15:24.893120287 +0100
@@ -742,7 +742,8 @@
     engine->resetTick();
 
     while (! (  SDL_QuitRequested()  //cout<<"catch close win"<<endl; //!rzr
-                || Keyboard::isKeyDown(SDLK_INSERT)))  {
+//               || Keyboard::isKeyDown(SDLK_INSERT)
+		))  {
 #if EM_DEBUG
       if (Keyboard::isKeyDown(SDLK_p)) {
         Keyboard::waitForKey();
