From: Adrian Bunk <bunk@debian.org>
Date: Fri, 31 Jul 2020 12:18:31 +0200
Subject: Fix the src/Makefile.am dependencies

---
 src/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 96768d3..0e2eaa3 100755
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -9,7 +9,7 @@ pinlib_LIBRARIES = libemilia_pin.a
 
 INCLUDES = -I../base -I../addon @INCLTDL@
 
-pinball_LDADD = ../src/libemilia_pin.a ../addon/libemilia_addon.a ../base/libemilia_base.a @LIBLTDL@
+pinball_LDADD = libemilia_pin.a ../addon/libemilia_addon.a ../base/libemilia_base.a @LIBLTDL@
 pinball_LDFLAGS = -export-dynamic
 
 pinball_SOURCES = Pinball.cpp
