From: Innocent De Marchi <tangram.peces@gmail.com>
Date: Fri, 31 Jul 2020 12:18:31 +0200
Subject: Some changes on old 03_desktop.patch.

Last-Update: 2018-04-04
---
 pinball.desktop | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/pinball.desktop b/pinball.desktop
index 52214d2..12d836d 100644
--- a/pinball.desktop
+++ b/pinball.desktop
@@ -1,12 +1,11 @@
 [Desktop Entry]
-Encoding=UTF-8
-Categories=Application;Game;ArcadeGame;X-Red-Hat-Extra
+Categories=Game;ArcadeGame;X-Red-Hat-Extra
 X-Desktop-File-Install-Version=0.2
 Name=Emilia Pinball
 Comment=OpenGL pinball game
 Exec=pinball
 Icon=pinball.xpm
-Terminal=0
+Terminal=false
 Type=Application
-                                                                                                                                                             
-                                                                                                                                                             
+Keywords=Game, ArcadeGame, Pinball
+                                                                                                                                                            
\ No newline at end of file
