From: Innocent De Marchi <tangram.peces@gmail.com>
Date: Fri, 31 Jul 2020 12:18:31 +0200
Subject: Load table game for default.

Bug-Debian: http://bugs.debian.org/555256
Last-Update: 2018-03-16
---
 src/Pinball.cpp | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/src/Pinball.cpp b/src/Pinball.cpp
index 9d89fcc..c98a4ed 100755
--- a/src/Pinball.cpp
+++ b/src/Pinball.cpp
@@ -771,8 +771,23 @@ int main(int argc, char *argv[]) {
           Table::getInstance()->getScore()->draw();
         }
         if (engine->getGroup(0) == NULL) {
-          EmFont::getInstance()->printRowCenter("no table loaded", 6);
-          EmFont::getInstance()->printRowCenter("press esc", 8);
+          //EmFont::getInstance()->printRowCenter("no table loaded", 6);
+          //EmFont::getInstance()->printRowCenter("press esc", 8);
+	srand (time(NULL));
+	int iTable ; 
+	iTable = rand() % 100 +1;
+	if (iTable < 51){
+		if (Table::getInstance()->loadLevel(engine, "tux") == 0) {
+      			Table::getInstance()->readHighScoresFile();	
+      			Table::getInstance()->getScore()->draw();
+    		}
+	     }
+	else {
+		if (Table::getInstance()->loadLevel(engine, "professor") == 0) {
+      			Table::getInstance()->readHighScoresFile();	
+      			Table::getInstance()->getScore()->draw();
+    		}
+	     }
         }
         engine->swap();
       }
