From: Innocent De Marchi <tangram.peces@gmail.com>
Date: Fri, 31 Jul 2020 12:18:30 +0200
Subject: Changed the storage directory of the high scores.

Last-Update: 2018-03-07
---
 src/Table.cpp | 37 +++++++++++++++++++++++--------------
 1 file changed, 23 insertions(+), 14 deletions(-)

diff --git a/src/Table.cpp b/src/Table.cpp
index 3cd8144..1976e2d 100755
--- a/src/Table.cpp
+++ b/src/Table.cpp
@@ -248,13 +248,18 @@ bool Table::readHighScoresFile() {
   m_mapHighScores.clear();
 
   //!rzr+ : fix w32
-  string sFileName =  m_sTableName + "/" + HIGH_SCORES_FILENAME;
-#ifdef RZR_PATHRELATIVE
-  sFileName = string( Config::getInstance()->getExeDir() )
-    +"/"+ m_sTableName +".cfg";
-#else
-  sFileName = string(EM_HIGHSCORE_DIR) + "/" + sFileName;
-#endif //!rzr-
+//  string sFileName =  m_sTableName + "/" + HIGH_SCORES_FILENAME;
+//#ifdef RZR_PATHRELATIVE
+//  sFileName = string( Config::getInstance()->getExeDir() )
+//    +"/"+ m_sTableName +".cfg";
+//#else
+//  sFileName = string(EM_HIGHSCORE_DIR) + "/" + sFileName;
+//#endif //!rzr-
+
+
+ char *home = getenv("HOME");
+ string sFileName = string(home? home:".") + "/.emilia/" + m_sTableName +
+    "."+HIGH_SCORES_FILENAME;
 
   ifstream file(sFileName.c_str());
   if (!file) {
@@ -309,13 +314,17 @@ bool Table::writeHighScoresFile() {
     return false;
   }
   //!rzr+ : fix w32
-  string sFileName =  m_sTableName + "/" + HIGH_SCORES_FILENAME;
-#ifdef RZR_PATHRELATIVE
-  sFileName = string( Config::getInstance()->getExeDir() )
-    +"/"+ m_sTableName +".cfg";
-#else
-  sFileName = string(EM_HIGHSCORE_DIR) + "/" + sFileName;
-#endif //!rzr-
+//  string sFileName =  m_sTableName + "/" + HIGH_SCORES_FILENAME;
+//#ifdef RZR_PATHRELATIVE
+//  sFileName = string( Config::getInstance()->getExeDir() )
+//    +"/"+ m_sTableName +".cfg";
+//#else
+//  sFileName = string(EM_HIGHSCORE_DIR) + "/" + sFileName;
+//#endif //!rzr-*/
+
+ char *home = getenv("HOME");
+ string sFileName = string(home? home:".") + "/.emilia/" + m_sTableName +
+    "."+HIGH_SCORES_FILENAME;
 
   ofstream file(sFileName.c_str());//, ios_base::out | ios_base::trunc);
   if (!file) {
